File: problems/1653_minimum_deletions_to_make_string_balanced/solution.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        b_count = 0
        result = 0
        for c in s:
            if c == 'b':
                b_count += 1
            else:
                result = min(result + 1, b_count)
        return result

File: problems/1653_minimum_deletions_to_make_string_balanced/Solution.java
class Solution {
    public int minimumDeletions(String s) {
        int bCount = 0, result = 0;
        for (char c : s.toCharArray()) {
            if (c == 'b') {
                bCount++;
            } else {
                result = Math.min(result + 1, bCount);
            }
        }
        return result;
    }
}
